03-rnaseq/gene-id-annotation_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,316 @@
+In the case of our [microarray gene identifier annotation example notebook](https://github.com/AlexsLemonade/refinebio-examples/tree/master/02-microarray/gene-id-annotation_microarray_01_ensembl.html), a Mouse (<i>Mus musculus</i>) dataset is used, meaning `org.Mm.eg.db` or `Mm` would also need to be used there.
+A full list of the annotation R packages from Bioconductor is at this [link](https://bioconductor.org/packages/release/BiocViews.html#___AnnotationData) [@annotation-packages].
+
+***
+
+<!-- Do not delete this line --> <a name="analysis" style="padding-top:56px;margin-top:-56px;">&nbsp;</a>
+
+# Obtaining Annotation for Ensembl IDs - RNA-seq
+
+Ensembl IDs can be used to obtain various different annotations at the gene/transcript level.
+Let's get ready to use the Ensembl IDs from our mouse dataset to obtain the associated Uniprot IDs.
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+In this analysis, we will be using the `org.Dr.eg.db` R package [@Carlson2019-zebrafish].
+
+```{r}
+# Install the Zebrafish package
+if (!("org.Dr.eg.db" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("org.Dr.eg.db", update = FALSE)
+}
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(org.Dr.eg.db)
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_SRP056136.tsv" # Replace with the name of your metadata file
+))
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "SRP056136.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+
+## Map Ensembl IDs to Uniprot IDs
```

Comment:
IDK if mapping to protein ids is a great idea. In general RNA and protein are pretty poorly correlated, so we may not want to encourage this science wise, though it is a thing you can do. 

What about RefSeq ids or another RNA-based id? 

Do the RefSeq IDs only have `NM_` accessions? and not `XM_` or etc?

Reply:
Let's do Entrez IDs - looks like #212 includes a symbols example. I expect these are going to be the most common use cases.

Reply:
I can try RefSeq again, but it was so memory intensive that it didn't complete locally on my machine. I can give it another try but I am not sure that's the way to go here.

Reply:
Perfect!